docs/authoring-components.md

```diff
@@ -2,51 +2,68 @@
 - List any reference to a CSS class, React component name, or ViewComponent name in the component's description.
-- Use [Figma variants](https://help.figma.com/hc/en-us/articles/360056440594-Create-and-use-variants) when authoring complex components
+- Use [Figma variants](https://help.figma.com/hc/en-us/articles/360056440594-Create-and-use-variants) for component states or variants/types. For example, a variant property could be for "State: rest, focus, hover, selected, disabled".
+- Use [Figma component properties](https://help.figma.com/hc/en-us/articles/5579474826519-Create-and-use-component-properties) to control the different options for hide/show layers (hide/show a leadingVisual), instance swapping (such as between an icon or avatar for a leadingVisual), and for text properties (headings, descriptions, etc).
 
 ## Variants
 
-In Primer, variants are used to quickly select different visual options to a component. 
+In Primer, variants and component properties are used to quickly select different states and/or visual options of a component. 
 
 ### Variant properties
 
-When creating a new component apply variant properties that follow existing parameters seen in other implementations where the component exsists, [Primer React](https://primer.style/react/) and [Primer ViewComponents](https://primer.style/view-components/). If a parameter exists but cannot be expressed in Figma (e.g. `alt` or `href`) it is not necessary to include.
+When creating a new component apply variant properties that follow existing parameters seen in other implementations where the component exsists, [Primer React](https://primer.style/react/) and [Primer ViewComponents](https://primer.style/view-components/). If a parameter exists but cannot be expressed in Figma (e.g. `alt` or `href`) it is not necessary to include. Variant properties will always appear above component properties in the properties panel and can only be reordered with other variant properties.
 
 ![Example: Avatar component in Figma](https://user-images.githubusercontent.com/10384315/131897226-b5253552-f13f-4e85-8226-d6ca2b34d938.png)
 
+### Component properties
+
+Certain visual options can be controlled with Figma's component properties. Apply component properties when controlling the visibility of a layer, swapping between specific instance options, or controlling the content of a text layer. Below is a list of common properties and the types of items they control. When ordering component properties, order them from the start to end to how they would be seen within the DOM. 
+
+As a reminder, variant properties will always appear above component properties in the properties panel.
+
+| Name | Description | Component property type |
+| :--: | :--: | :--: |
+| ⬅️ visual? | Show/hide the leading visual which is a the start of the component (left-side) | [Boolean](https://help.figma.com/hc/en-us/articles/5579474826519-Create-and-use-component-properties#h_01G2Q5GA6DEB604H2E5H5C5TA4) |
```

Comment:
In this example there's both `⬅️ visual` for `leadingVisual`, and `leadingAction` in full. Should we normalize this with the same props used in code? Such as: `leadingVisual` and `leadingAction`.

@ashygee after we chatted about this, I tested using both "⬅️" and `<-` (which is nicely rendered by Figma as "←"), and I found that no prefix gives the maximum legibility in such a tight space:

| Prefix | Result |
|--------|--------|
| ⬅️ | ![Rendering](https://user-images.githubusercontent.com/293280/168685370-26ff7b70-005d-48f0-a9ad-3642d55b0985.png) |
| `<-` | ![Rendering](https://user-images.githubusercontent.com/293280/168685373-65d0e140-079b-4571-ba37-184f306a1b19.png) |
| (None) | ![Rendering](https://user-images.githubusercontent.com/293280/168685420-996933af-6c52-4eb3-b4bb-c20a82741d15.png) |

Reply:
We can remove them and add to the layer name only as suggested by @langermank in her comment in https://github.com/github/primer/issues/979#issuecomment-1128370017.